improc/simulator.py

```diff
@@ -453,6 +501,331 @@ def get_star_flux_values(self):
         return self.star_mean_fluxes
 
 
+class SimGalaxies:
+    """
+    Container for the properties of a set of simulated galaxies.
+
+    Galaxies here are a very simple object,
+    a bulge with a Sersic profile, and a disk with an exponential profile.
+    The bulge is spherically symmetric, while the disk is very thin
+    and has a uniformly chosen cos(i) inclination.
+    The ratio of total flux in the bulge vs disk is chosen randomly
+    (from???)
+    as is the ratio of the bulge length scale vs. the disk length scale.
+    (we need to figure out if the ranges for these ratios are fixed or
+    a user-defined parameter).
+
+    """
+    def __init__(self):
+        self.galaxy_number = None  # average number of galaxies in each field
+
+        self.galaxy_min_flux = None  # minimal flux for the galaxy power law
+        self.galaxy_flux_power_law = None  # power law index of the flux of galaxies
+        self.galaxy_mean_fluxes = None  # for each galaxy, the mean flux (in photons per total exposure time)
+
+        self.galaxy_mean_x_pos = None  # for each galaxy, the mean x position
+        self.galaxy_mean_y_pos = None  # for each galaxy, the mean y position
+        self.galaxy_position_std = None  # for each galaxy, the variation in position (in both x and y)
+
+        self.galaxy_min_width = None  # minimal width for the galaxy power law
+        self.galaxy_width_power_law = None  # power law index of the width of galaxies
+        self.galaxy_width_values = None  # for each galaxy, the width (in pixels)
+
+        self.galaxy_cos_is = None  # for each galaxy, the cos(inclination), 1 means face on, 0 means edge on
+        self.galaxy_rotations = None  # for each galaxy, the position angle (rotation) from North, towards East, in deg
+
+        self.galaxy_sersic_flux_ratios = None  # the ratio of sersic flux to exponential flux for each galaxy
+        self.galaxy_sersic_scale_ratios = None  # the length scale of the sersic profile relative to width, per galaxy
+        self.galaxy_exp_scale_ratios = None  # the length scale of the exponential profile relative to width, per galaxy
+
+    def make_galaxy_list(self, imsize):
+        """
+        Make a field of galaxies.
+        Uses the power law to draw random mean fluxes
+        and widths, and uniform orientations (both in inclination
+        and in position angle) and positions.
+        The input, imsize, is a tuple of (imsize_x, imsize_y).
+        """
+        rng = np.random.default_rng()
```

Comment:
Here and elsewhere, it might be worth having an optional "rng" parameter, and using np.random.default_rng() if that parameter is none.  That would allow us to use a pre-seeded random number generator in order to make tests deterministic.